### PR TITLE
fix: check for `is-marketer` and `is-sales-admin` keys in data

### LINKED ISSUE #5644: deleting Inactive Users assigning them roles automatically
**Describe the bug**

Deleting Inactive Users assigning them roles automatically, please see the GIF for clarity

**Expected behavior**

No such roles should be assigned to any deleted user

**GIF**

ezgif com-video-to-gif 5

**Desktop (please complete the following information):**
 - OS: [e.g. iOS] Ubuntu 16.04LTS
 - Browser [e.g. chrome, safari] Chrome

**Additional context**

Working on it

### PR DESCRIPTION
**Pull request by uds5501**
Fixes #5644 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

#### Short description of what this resolves:
makes sure that `is-marketer` and `is-sales-admin` properties aren't altered upon record deletion

#### Changes proposed in this pull request:
image

- in delete route, only `deleted-at` is present hence the condition is proved to be false and the inversion happened

### CODE REVIEW

**Comment by uds5501** [comment]
@iamareebjamal @CosmicCoder96 Review please! :)

File: app/api/users.py [hunk 1]
```diff
@@ -216,10 +216,10 @@ def before_update_object(self, user, data, view_kwargs):
         if has_access('is_super_admin') and data.get('is_admin') and data.get('is_admin') != user.is_admin:
             user.is_admin = not user.is_admin
 
-        if has_access('is_admin') and data.get('is_sales_admin') != user.is_sales_admin:
+        if has_access('is_admin') and data.get('is_sales_admin') != user.is_sales_admin and ('is_sales_admin' in data):
```
Comment (iamareebjamal) [inline app/api/users.py]: Your check should happen before getting the value

**Comment by uds5501** [comment]
@iamareebjamal will these do?

File: app/api/users.py [hunk 2]
```diff
@@ -216,11 +216,13 @@ def before_update_object(self, user, data, view_kwargs):
         if has_access('is_super_admin') and data.get('is_admin') and data.get('is_admin') != user.is_admin:
             user.is_admin = not user.is_admin
 
-        if has_access('is_admin') and data.get('is_sales_admin') != user.is_sales_admin and ('is_sales_admin' in data):
-            user.is_sales_admin = not user.is_sales_admin
+        if 'is_sales_admin' in data:
+            if has_access('is_admin') and data.get('is_sales_admin') != user.is_sales_admin:
```
Comment (iamareebjamal) [inline app/api/users.py]: No need to use nested if, just use and operator

Comment (uds5501) [reply app/api/users.py]: Like in current commit?

**Pull request merged**